### ISSUE TITLE
How to use Bootstrap when refitting final stage?

### ISSUE THREAD

**Issue by federiconuta** [microsoft/EconML#339]
Hi all,

So basically, roughly I have:

```
cate = DMLIV(model_Y_X(), model_T_X(), model_T_XZ(),
             dmliv_model_effect(), dmliv_featurizer(),
             n_splits= 20)

##### defining some variable in here
cate.fit(Y, T[:,0], XW, Z, store_final=True)

##### choosing the preferred subset of XW to refit the final in here
cate.refit_final(ph_dmliv_model_effect(), dmliv_featurizer())
```

So first of all. Should I use BootstrapEstimator in both or just when refitting the final stage as I am guessing (since the final estimates are on the refitted final stage). Secondly, when doing: 

`BootstrapEstimator(cate, n_bootstrap_samples=100).refit_final(ph_dmliv_model_effect(), dmliv_featurizer())`

the following error appears: "unsupported operand type(s) for +: 'DMLIV' and 'DMLIV'" after having finished the replicates.

What's going wrong here? Seems like I have to call an attribute of the class DMLIV like fit or refit, but I am not really sure...
Moreover, is the refitting to be done with the training set? That's because I am specifying fold0, fold1, fold00, folld11 of issue #94 in final model, but I don't know how Bootstrap acts. Moreover, by doing:

```
boot_est=BootstrapEstimator(est,n_bootstrap_samples=100)
te_pred_interval = boot_est.effect_interval(X_test, lower=1, upper=99)
te_pred_interval
```

the intervals turn out to be equal to the estimates and equal among each other.

Thank you again!

**Comment by kbattocchi** [comment]
Please keep in mind that generally the prototypes are unsupported.  We'll be adding refit support to all of our OrthoLearner subclasses in our next release, so your best bet is probably to wait a week and then use the supported method for achieving this.  This is being tracked by #310.

**Comment by federiconuta** [comment]
Ok thank you!

**Comment by federiconuta** [comment]
@kbattocchi  could you please inform us when this enhancement is ready? (Or otherwise, where can I see if it is and git clone the leaf?) Thank you so much!

**Comment by federiconuta** [comment]
Should I just look at #310?

**Comment by kbattocchi** [comment]
Yes; we've run into an unrelated issue with some other work that has delayed us slightly, but I hope we'll have a beta release (0.9.0b1) our very soon that handles this.  #310 is the best place to look to see when that's done.

**Comment by federiconuta** [comment]
@kbattocchi  thank you for the prompt answer. Thank you very much for your work!

**Comment by federiconuta** [comment]
@kbattocchi. Hi. Is there any news on #310? :)

**Comment by kbattocchi** [comment]
@federiconuta Apologies - we ran into an issue with one of our dependencies that made us unable to release before the winter holiday period here in the US.  We're now aiming for this upcoming Friday, January 8 instead.  Sorry for any inconvenience.

**Comment by federiconuta** [comment]
@kbattocchi  no need to apology at all. Thank you for your efforts! So I will check by Friday the 8th issue #310. Just if you please could let us know if any delay intervene would be great. Thank you again for your work!

**Comment by kbattocchi** [comment]
Sure.  But just to set expectations, we've got several big changes going in so I think it's likely to take us up to the end of the day on Friday our time (we're mostly based on the East Coast of the US), so if you're in Europe then it will probably be quite late your time before we get the update out.

**Comment by kbattocchi** [comment]
Hi @federiconuta - we're still waiting on an update to one of the python libraries we depend on to put out the new release.  Our target is now Monday, January 11.  Sorry again for the delay.

**Comment by federiconuta** [comment]
Hi @kbattocchi. Thank you for the update. No problem

**Comment by federiconuta** [comment]
@kbattocchi hi. Is there any news on the topic? Moreover could you please explain where we could find examples on that? 

Thank you very much!

**Comment by kbattocchi** [comment]
Hi @federiconuta, yes, we've just created a new release - please give it a shot.  It's a pre-release, so try `pip install -U --pre econml`.  We've had to change how we create the package, so please if you have any issues with installation please open a new issue with the details.

Please note that this release makes several fairly large structural changes: for example, our IV estimators are now all nested under `econml.iv` (e.g. `econml.iv.dml.DMLIV`).  Our estimators which use a two-stage cross-fitting approach now all support refitting just the final model by calling `refit_final()` - there are a few examples of this in our [Double Machine Learning Examples](https://github.com/microsoft/EconML/blob/master/notebooks/Double%20Machine%20Learning%20Examples.ipynb) notebook.  Please let us know if you run into any issues.
